app/models/exporters/csv_exporter.rb

```diff
@@ -0,0 +1,36 @@
+require 'csv'
+
+module Exporters
+  class UnknownExporter < StandardError; end
+
+  class CsvExporter
+    def dump(workflow_id)
+      path = get_path(workflow_id)
+      items = get_items(workflow_id)
+
+      model_cols = get_model_cols
+      json_cols = get_unique_json_cols(workflow_id)
+
+      CSV.open(path, "wb",
+        :write_headers => true,
+        :headers => get_csv_headers(workflow_id)) do |csv|
+
+        items.each do |item|
```

Comment:
Can you do a `.find_each` instead here? That'll fetch stuff in small batches so that it doesn't run out of memory on large datasets. You can merge it after that, LGTM.